### DISCUSSION TITLE
Stabilising Linux Builds

### DISCUSSION THREAD

**Discussion by mikeskydev** [icosa-gallery/open-brush#215] [General]
Making a thread to discuss testing of Linux + SteamVR Builds. It would be great to get it on the Steam store, but we're lacking in testers and people familiar with Linux VR at the moment.

Possibly relevant issues:
#67 
#138

Most recent builds available direct on GitHub:
https://github.com/icosa-gallery/open-brush/releases/tag/1.0.0
Or itch:
https://openbrush.itch.io/openbrush

**Comment by jrtberlin** [comment]
Hi, I'm able to test things with SteamVR on Arch Linux. Is there a beta branch on Steam that I'm not aware of?

**Reply by mikeskydev** [reply]
Hey, thanks for offering! Not yet, but we could look at setting that up. How was your luck with the build from https://github.com/icosa-gallery/open-brush/releases/tag/1.0.0 or itch.io?

**Reply by jrtberlin** [reply]
The build seems to have good performance overall but crashes when I click the `Audio Reactor` icon on the left controller panel.

Build: OpenBrush Beta from itch.io
Linux: 5.16.2
Nvidia Driver: 495.46
SteamVR: beta 1.21.6

```
 ./OpenBrush          
Set current directory to /home/jrt/.config/itch/apps/openbrush
Found path: /home/jrt/.config/itch/apps/openbrush/OpenBrush
[1]    91529 abort (core dumped)  ./OpenBrush

```

After the crash, restarting `OpenBrush` results in a crash, even after deleting `~/Documents/OpenBrush`. 
Successful reopening requires a restart of SteamVR.

The[ log of the crash](https://bin.jrtberlin.de/?8232fe39c6e4e01f#HHiqYu81ZfwGJ6AAFyu5PEdzBxkZj6UpKzTC89YQeF8n) seems to relate to the Audio Reactor function:
```
#12 0x00000041cc73d8 in CSCore.CoreAudioAPI.MMDeviceEnumerator:CreateMmDeviceEnumerator ()
```

**Reply by mikeskydev** [reply]
Thanks for the logs, this really helps! Yeah I'm guessing that the system used for capturing audio wasn't designed to work for linux. We can probably patch that fairly quickly to at least not crash the app, getting audio reactivity working might take a bit longer though.

**Reply by jrtberlin** [reply]
Let me know if you need any specific testing.

**Comment by mikeskydev** [comment]
We are using OpenXR in v2.0, which unfortunately Unity has not shipped Linux support for yet. Please join us in raising an issue with Unity when we release 2.0!

We're also interested in testing the Windows OpenXR version through Wine/Proton, as I've heard this is possible, but we don't have much experience in this area. We'd love some feedback and testing in that regard.
